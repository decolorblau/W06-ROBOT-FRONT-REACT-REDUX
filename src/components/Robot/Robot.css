.image-container {
  max-height: 300px;
  max-width: 100%;
  height: 100%;
  width: 400px;
  overflow-x: hidden;
}

.image-robot {
  height: 100%;
  width: 100%;
  object-fit: cover;
}
