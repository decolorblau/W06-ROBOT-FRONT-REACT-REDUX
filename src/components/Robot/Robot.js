const Robot = ({ robot }) => {
  return (
    <div className="card" style="width: 18rem;">
      <img src={robot.imageUrl} className="card-img-top" alt="robot" />
      <div className="card-body">
        <h3 className="card-title">{robot.name}</h3>
        <p className="card-text">Speed: {robot.features.speed}</p>
        <p className="card-text">Resistance: {robot.features.resistance}</p>
        <p className="card-text">
          Year of Creation: {robot.features.yearCreation}
        </p>
        <button className="btn btn-primary">Update</button>
        <button className="btn btn-primary">Delete</button>
      </div>
    </div>
  );
};

export default Robot;
