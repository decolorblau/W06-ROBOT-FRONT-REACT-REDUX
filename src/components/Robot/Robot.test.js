import { Provider } from "react-redux";
import Robot from "./Robot";
import ReactTestRenderer from "react-test-renderer";
import configureStore from "../../redux/store";

describe("Given a Task component", () => {
  describe("When it receives and object with task", () => {
    test("then it should render a snapshot of task", () => {
      const robot = {
        id: 1,
        name: "robot",
        imageUrl:
          "https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.hobbyconsolas.com%2Fnoticias%2Fsorprendente-nueva-patente-sony-robot-nos-acompana-jugar-ps5-623609&psig=AOvVaw2Q1mBWnM8eXuUvp8J6IhSV&ust=1636372563694000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCIDEg_-YhvQCFQAAAAAdAAAAABAD",
        features: {
          speed: 9,
          resistance: 4,
          yearCreation: 2020,
        },
      };

      const store = configureStore();

      const renderRobot = ReactTestRenderer.create(
        <Provider store={store}>
          <Robot robot={robot} />
        </Provider>
      );

      expect(renderRobot.toJSON()).toMatchSnapshot();
    });
  });
});
