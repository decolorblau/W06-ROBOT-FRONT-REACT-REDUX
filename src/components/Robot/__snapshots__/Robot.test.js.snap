// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Task component When it receives and object with task then it should render a snapshot of task 1`] = `
<div
  className="card"
>
  <div
    className="card-img-top image-container"
  >
    <img
      alt="robot"
      className="card-img-top image-robot"
      src="https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.hobbyconsolas.com%2Fnoticias%2Fsorprendente-nueva-patente-sony-robot-nos-acompana-jugar-ps5-623609&psig=AOvVaw2Q1mBWnM8eXuUvp8J6IhSV&ust=1636372563694000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCIDEg_-YhvQCFQAAAAAdAAAAABAD"
    />
  </div>
  <div
    className="card-body"
  >
    <h3
      className="card-title"
    >
      robot
    </h3>
    <p
      className="card-text"
    >
      Speed: 
      9
    </p>
    <p
      className="card-text"
    >
      Resistance: 
      4
    </p>
    <p
      className="card-text"
    >
      Year of Creation: 
      2020
    </p>
    <button
      className="btn btn-outline-primary"
    >
      Update
    </button>
    <button
      className="btn btn-outline-danger m-1"
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Given a robot component When it receives and object with robot then it should render a snapshot of robot 1`] = `
<div
  className="card"
>
  <div
    className="card-img-top image-container"
  >
    <img
      alt="robot"
      className="card-img-top image-robot"
      src="https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.hobbyconsolas.com%2Fnoticias%2Fsorprendente-nueva-patente-sony-robot-nos-acompana-jugar-ps5-623609&psig=AOvVaw2Q1mBWnM8eXuUvp8J6IhSV&ust=1636372563694000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCIDEg_-YhvQCFQAAAAAdAAAAABAD"
    />
  </div>
  <div
    className="card-body"
  >
    <h3
      className="card-title"
    >
      robot
    </h3>
    <p
      className="card-text"
    >
      Speed: 
      9
    </p>
    <p
      className="card-text"
    >
      Resistance: 
      4
    </p>
    <p
      className="card-text"
    >
      Year of Creation: 
      2020
    </p>
    <button
      className="btn btn-outline-primary"
    >
      Update
    </button>
    <button
      className="btn btn-outline-danger m-1"
    >
      Delete
    </button>
  </div>
</div>
`;
